src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -344,12 +349,17 @@ private static Uri CreateConnectUrl(Uri url, NegotiationResponse negotiationResp
                     "Cannot send messages when the connection is not in the Connected state.");
             }
 
+            /* Gross - send as binary HttpRequest can handle that
```

Comment:
The problem is that the first message sent by HubConnection is supposed to be only text but HttpConnection will convert everything.